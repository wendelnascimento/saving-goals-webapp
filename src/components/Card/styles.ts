import styled from 'styled-components';

export const CardContainer = styled.div`
  background: ${({ theme }) => theme.white};
  padding: 32px;
  border-radius: 4px;
  border: 1px solid ${({ theme }) => theme.lightGray};

  @media screen and (max-width: 550px) {
    border-top-left-radius: 20px;
    border-top-right-radius: 20px;
    border: none;
  }
`;
