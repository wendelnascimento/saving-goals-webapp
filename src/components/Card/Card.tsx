import React from 'react';

import { CardContainer } from './styles';

const Card: React.FC = ({ children }) => (
  <CardContainer>{children}</CardContainer>
);

export default Card;
