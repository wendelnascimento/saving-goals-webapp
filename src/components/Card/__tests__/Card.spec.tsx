import React from 'react';
import { shallow, mount } from 'enzyme';

import Card from '../Card';

describe('<Card />', () => {
  it('Should render', () => {
    shallow(<Card />);
  });

  it('Should render with children', () => {
    shallow(
      <Card>
        <h1>Test</h1>
      </Card>,
    );
  });

  it('Should math stored snapshot', () => {
    const wrapperWithChildren = mount(
      <Card>
        <h1>Test</h1>
      </Card>,
    );
    const emptyWrapper = mount(<Card />);

    expect(wrapperWithChildren).toMatchSnapshot();
    expect(emptyWrapper).toMatchSnapshot();
  });
});
