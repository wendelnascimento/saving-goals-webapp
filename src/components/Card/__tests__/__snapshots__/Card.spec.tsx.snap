// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Should math stored snapshot 1`] = `
.c0 {
  padding: 32px;
  border-radius: 4px;
  border: 1px solid;
}

@media screen and (max-width:550px) {
  .c0 {
    border-top-left-radius: 20px;
    border-top-right-radius: 20px;
    border: none;
  }
}

<Card>
  <styled.div>
    <div
      className="c0"
    >
      <h1>
        Test
      </h1>
    </div>
  </styled.div>
</Card>
`;

exports[`<Card /> Should math stored snapshot 2`] = `
.c0 {
  padding: 32px;
  border-radius: 4px;
  border: 1px solid;
}

@media screen and (max-width:550px) {
  .c0 {
    border-top-left-radius: 20px;
    border-top-right-radius: 20px;
    border: none;
  }
}

<Card>
  <styled.div>
    <div
      className="c0"
    />
  </styled.div>
</Card>
`;
